README.md

```diff
@@ -0,0 +1,180 @@
+
+  # If the node should wait for `deploy` to connect for a second time after activation, to confirm the server has not been ruined.
+  # This defaults to `false`, though it is strongly recommend you activate it if you value safety
+  magicRollback = true;
+
+  # The path which deploy-rs will use for temporary files, this is currently only used by `magicRollback` to create an inotify watcher in
+  # If not specified, this will default to `/tmp/deploy-rs`
+  # (if `magicRollback` is in use, this _must_ be writable by `user`)
+  tempPath = "/home/someuser/.deploy-rs";
+}
+```
+
+### Putting it together
+
+`deploy-rs` is designed to be used with Nix flakes (this currently requires an unstable version of Nix to work with). There is a Flake-less mode of operation which will automatically be used if your available Nix version does not support flakes, however you will likely want to use a flake anyway, just with `flake-compat` (see [this wiki page](https://nixos.wiki/wiki/Flakes) for usage).
+
+`deploy-rs` also outputs a `lib` attribute, with tools used to make your definitions simpler and safer, including `deploy-rs.lib.${system}.setActivate` (see prior section "Profile"), and `deploy-rs.lib.${system}.deployChecks` which will let `nix flake check` ensure your deployment is defined correctly.
+
+A basic example of a flake that works with `deploy-rs` and deploys a simple NixOS configuration could look like this
+
+```nix
+{
+  description = "Deployment for my server cluster";
+
+  # For accessing `deploy-rs`'s utility Nix functions
+  inputs.deploy-rs.url = "github:serokell/deploy-rs";
+
+  outputs = { self, nixpkgs, deploy-rs }: {
+    nixosConfigurations.some-random-system = nixpkgs.lib.nixosSystem {
+      system = "x86_64-linux";
+      modules = [ ./some-random-system/configuration.nix ];
+    };
+
+    deploy.nodes.some-random-system.profiles.system = {
+        user = "root";
+        path = deploy-rs.lib.x86_64-linux.setActivate self.nixosConfigurations.some-random-system.config.system.build.toplevel "./bin/switch-to-configuration switch";
+    };
+  };
+
+  # This is highly advised, and will prevent many possible mistakes
+  checks = builtins.mapAttrs (system: deployLib: deployLib.deployChecks self.deploy) deploy-rs.lib;
+}
+```
+
+There are full working deploy-rs Nix expressions in the [examples folder](./examples), and there is a JSON schema [here](./interface.json) which is used internally by the `deployChecks` mentioned above to validate your expressions.
+
+## Idea
+
+`deploy-rs` is a simple Rust program that will take a Nix flake and use it to deploy any of your defined profiles to your nodes. This is _strongly_ based off of [serokell/deploy](https://github.com/serokell/deploy), designed to replace it and expand upon it.
+
+This type of design (as opposed to more traditional tools like NixOps or morph) allows for lesser-privileged deployments, and the ability to update different things independently of eachother.
+
+## Things to work on
```

Comment:
Why is this here instead of youtrack?

Reply:
Not instead of, in addition to, this is a public project so hence the documentation is not only for us, it can give other people an understanding of the direction of the project

Reply:
In the case of public open source projects (that we expect anyone else to actually care about), I believe our policy is to use github issues instead of youtrack.